### ISSUE TITLE
Not compatible with Gradle 8.0

### ISSUE THREAD

**Issue by masarbazi** [cafebazaar/react-native-poolakey#15]
Using the latest RN version (0.72 for now), this package is not compatible with Gradle 8.0 and building the project causes the following error:

```
error 
Failed to install the app. 
Command failed with exit code 1: gradlew.bat tasks FAILURE: Build failed with an exception. 

* What went wrong:
A problem occurred configuring project ':cafebazaar_react-native-poolakey'. 
> Failed to notify project evaluation listener. 
> org.gradle.api.GradleException: Adding a Configuration as a dependency is no longer allowed as of Gradle 8.0. 
> KotlinJvmAndroidCompilation with name 'debug' not found.
  
* Try: > Run with --stacktrace option to get the stack trace. > Run with --info or --debug option to get more log output. > Run with --scan to get full insights. * Get more help at https://help.gradle.org BUILD FAILED in 1m 18s
> Task :gradle-plugin:pluginDescriptors > Task :gradle-plugin:processResources > Task :gradle-plugin:compileKotlin > Task :gradle-plugin:compileJava NO-SOURCE > Task :gradle-plugin:classes > Task :gradle-plugin:jar > Task :gradle-plugin:inspectClassesForKotlinIC 5 actionable tasks: 5 executed.
```

**Comment by sepehrsohrabii** [comment]
Apparently, the GitHub repo is up to date but the npm package is not. So you should install the package directly from GitHub repository. Using the bellow command solved the problem for me.
`npm install git+https://github.com/cafebazaar/react-native-poolakey.git`

**Comment by masarbazi** [comment]
Thanks Sepehr, I could resolve previous error, however now I'm getting another one. 

```
* What went wrong:
Execution failed for task ':tasks'.
> Could not create task ':cafebazaar_react-native-poolakey:compileDebugAndroidTestKotlin'.
   > Cannot use @TaskAction annotation on method AbstractKotlinCompile.execute() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.
```
What react native version you're using?

**Comment by sepehrsohrabii** [comment]
You're welcome.
I think these kinds of errors will be fixed by removing the Android folder and again building the project.
And about the React native version, I'm using the last one.

**Comment by soroshzzz26** [comment]
same issue here

> Thanks Sepehr, I could resolve previous error, however now I'm getting another one.
> 
> ```
> * What went wrong:
> Execution failed for task ':tasks'.
> > Could not create task ':cafebazaar_react-native-poolakey:compileDebugAndroidTestKotlin'.
>    > Cannot use @TaskAction annotation on method AbstractKotlinCompile.execute() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.
> ```
> 
> What do you think this is related to? What react native version you're using?


same issue here

**Comment by soroshzzz26** [comment]
i fixed the issue by upgrading kotlin version to 1.8.10 manually 
you could fix this by forking like me
i dont think they accepting the pr 
they had only one job to update project correctly and test the build 😂

**Comment by barghi** [comment]
as @sepehrsohrabii said, install the package with
`yarn add git+https://github.com/cafebazaar/react-native-poolakey.git`

then add kotlinVersion to your `android/build.gradle`

```
buildscript {
    ext {
        ...
        
	kotlinVersion = "1.8.0"
	
        ...
    }
   ...
}
```

**Comment by ahmousavi** [comment]
The problem fixed in version 3.1.0 of the package. If you are using an older version of React Native and have any problem in using version 3 you can use version 2.2.0 of the package as it is compatible with older React Native versions.
